lib/web/app/transport.go

```diff
@@ -159,39 +165,62 @@ func (t *transport) rewriteRequest(r *http.Request) error {
 	return nil
 }
 
-// websocketsDialer returns a function that dials a websocket connection
-// over the transport's reverse tunnel.
-func websocketsDialer(tr *http.Transport) forward.Dialer {
-	return func(network, address string) (net.Conn, error) {
-		conn, err := tr.DialContext(context.Background(), network, address)
+// DialContext
+func (t *transport) DialContext(ctx context.Context, _ string, _ string) (net.Conn, error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
```

Comment:
I've changed the servers list to a `sync.Map` (since it performs write operations) to reduce lock contention, and now the dials won't be done during locks.